core/server/api_container/server/service_network/default_service_network.go

```diff
@@ -1445,24 +1444,31 @@ func (network *DefaultServiceNetwork) destroyService(ctx context.Context, servic
 		},
 		Statuses: nil,
 	}
-	_, failedToDestroyUuids, err := network.kurtosisBackend.DestroyUserServices(context.Background(), network.enclaveUuid, userServiceFilters)
+	successfullyDestroyedUuids, failedToDestroyUuids, err := network.kurtosisBackend.DestroyUserServices(context.Background(), network.enclaveUuid, userServiceFilters)
 	if err != nil {
-		errorResult = stacktrace.Propagate(err, "Attempted to destroy the services with UUID '%v' but had no success. You must manually destroy the services. Kurtosis will now try to remove its sidecar if it exists but might it fail as well.", serviceUuid)
+		return stacktrace.Propagate(err, "Attempted to destroy the services with UUID '%v' but had no success. You must manually destroy the services. Kurtosis will now try to remove its sidecar if it exists but might it fail as well.", serviceUuid)
 	}
-	if failedToDestroyErr, found := failedToDestroyUuids[serviceUuid]; found {
-		errorResult = stacktrace.Propagate(failedToDestroyErr, "Attempted to destroy the services with UUID '%v' but had no success. You must manually destroy the services. Kurtosis will now try to remove its sidecar if it exists but might it fail as well.", serviceUuid)
+	if _, found := successfullyDestroyedUuids[serviceUuid]; found {
+		if len(failedToDestroyUuids) != 0 {
+			return stacktrace.NewError("Service '%s' was successfully destroyed but unexpected error were returned by Kurtosis backend. This is unexpected and should be investigated. Here are the errors:\n%v", serviceUuid, failedToDestroyUuids)
+		}
+	} else {
+		if failedToDestroyErr, found := failedToDestroyUuids[serviceUuid]; found {
```

Comment:
randomly being paranoid; but if the failedToDestroyErr is nil then the propagate at line 1457 will panic.

https://go.dev/play/p/pHL11xwBzE1

^ just confirming whether a dictionary can map to a nil; it can.

Reply:
Wait... But if found is true, it means there is a value associated to it. This is the contract for `DestroyUserServices`, and if this function can return an error map with a key associated with `nil`, then we should fix this function rather than handling it here; I checked it cannot